src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -89,12 +93,20 @@ private void addReject(long val) {
      * @param excludeValues   The terms to be excluded
      *                  (may only be {@code null} if one of the other arguments is none-null.
      */
-    public IncludeExclude(Pattern include, Pattern exclude, Set<BytesRef> includeValues, Set<BytesRef> excludeValues) {
-        assert includeValues != null || include != null || 
+    public IncludeExclude(RegExp include, RegExp exclude, Set<BytesRef> includeValues, Set<BytesRef> excludeValues) {
```

Comment:
Another option for these is to reverse-lookup their ordinals against the dv instance, and you just exclude those ordinals. This should make this option much faster as it wouldnt need to traverse the Terms at all and do contains(BytesRef)